cloud/scope/powervs_cluster.go

```diff
@@ -1198,42 +1210,25 @@ func (s *PowerVSClusterScope) createVPCSubnet(subnet infrav1beta2.Subnet) (*stri
 	if resourceGroupID == "" {
 		return nil, fmt.Errorf("failed to fetch resource group ID for resource group %v, ID is empty", s.ResourceGroup())
 	}
-	var zone string
-	if subnet.Zone != nil {
-		zone = *subnet.Zone
-	} else {
-		vpcZones, err := regionUtil.VPCZonesForVPCRegion(*s.VPC().Region)
-		if err != nil {
-			return nil, err
-		}
-		// TODO(karthik-k-n): Decide on using all zones or using one zone
-		if len(vpcZones) == 0 {
-			return nil, fmt.Errorf("failed to fetch VPC zones, error: %v", err)
-		}
-		zone = vpcZones[0]
-	}
 
 	// create subnet
 	vpcID := s.GetVPCID()
 	if vpcID == nil {
 		return nil, fmt.Errorf("VPC ID is empty")
 	}
-	cidrBlock, err := s.IBMVPCClient.GetSubnetAddrPrefix(*vpcID, zone)
```

Comment:
You also need to remove the function definitions as well, if they are not being used anywhere else.

Reply:
Removed this function definition and declaration since its not used anywhere else